src/main/java/org/rsa/command/commands/CompileCommand.java

```diff
@@ -0,0 +1,147 @@
+                    CompilerInfo::getLanguage,
+                    Collectors.mapping(
+                            CompilerInfo::getVersion, Collectors.toSet()
+                    )
+            ));
+
+    public CompileCommand() {
+        super("compile", "Compiles code.");
+        addOptionData(new OptionData(OptionType.STRING, "language",
+                "The language to compile your code in.", true, true));
+        addOptionData(new OptionData(OptionType.STRING, "version",
+                "The version of the language to use.", true, true));
+        addOptionData(new OptionData(OptionType.STRING, "code",
+                "The code to compile.", true));
+        addOptionData(new OptionData(OptionType.BOOLEAN, "display",
+                "Display the result to the channel.", false));
+        setIsAutocomplete();
+    }
+
+    @Override
+    public void onCommandAutoCompleteInteraction(@NotNull CommandAutoCompleteInteractionEvent event) {
+        String focusedOption = event.getFocusedOption().getName();
+        Set<String> setToFilter = Collections.emptySet();
+
+        if (focusedOption.equals("language")) {
+            setToFilter = languageList;
+        } else if(focusedOption.equals("version")) {
+            OptionMapping languageOption = event.getOption("language");
+
+            if (languageOption != null && languageVersions.containsKey(languageOption.getAsString())) {
+                setToFilter = languageVersions.get(languageOption.getAsString());
+            }
+        }
+
+        List<Command.Choice> options = getFilteredOptions(event, setToFilter);
+
+        event.replyChoices(options).queue();
+    }
+
+    private @NotNull List<Command.Choice> getFilteredOptions(@NotNull CommandAutoCompleteInteractionEvent event,
+                                                             Set<String> setToFilter) {
+        return setToFilter.stream()
+                .filter(e -> e.startsWith(event.getFocusedOption().getValue()))
+                .map(e -> new Command.Choice(e, e))
+                .limit(25)
+                .toList();
+    }
+
+    @Override
+    public void onSlashCommandInteraction(@NotNull SlashCommandInteractionEvent event) {
+        // TODO: Ship to wandbox.
+        OptionMapping languageMapping = event.getOption("language");
+        OptionMapping versionMapping = event.getOption("version");
+        OptionMapping codeMapping = event.getOption("code");
+
+        String language;
+        String version;
+        String code;
+
+        if (languageMapping != null) {
+            language = languageMapping.getAsString();
+        } else {
+            language = null;
+        }
+
+        if (versionMapping != null) {
+            version = versionMapping.getAsString();
+        } else {
+            version = null;
+        }
+
+        if (codeMapping != null) {
+            code = codeMapping.getAsString();
+        } else {
+            code = null;
+        }
```

Comment:
1. Mapping -> Value done this way to simulate "effectively final" variables for Lambda use below.
2. Should introduce an OptionMappingToValue<T> method.